.github/workflows/annotate.yml

```diff
@@ -0,0 +1,58 @@
+name: Annotate checks
+
+on:
+  workflow_run:
+    workflows: ["ci", "docs"]
+    types:
+      - completed
+
+jobs:
+  annotate:
+    if: github.repository_owner == 'trinodb' && github.event.workflow_run.event == 'pull_request'
+    runs-on: ubuntu-latest
+    steps:
+      - name: 'Download artifact'
+        uses: actions/github-script@v5
+        with:
+          script: |
+            const fs = require('fs');
+            const opts = github.rest.actions.listWorkflowRunArtifacts.endpoint.merge({
+               owner: context.repo.owner,
+               repo: context.repo.repo,
+               run_id: ${{github.event.workflow_run.id }},
+            });
+            const artifacts = await github.paginate(opts);
+            for (const artifact of artifacts) {
+              if (!artifact.name.startsWith('test report ')) {
+                continue;
+              }
+              const download = await github.rest.actions.downloadArtifact({
+                 owner: context.repo.owner,
+                 repo: context.repo.repo,
+                 artifact_id: artifact.id,
+                 archive_format: 'zip',
+              });
+              fs.writeFileSync('${{github.workspace}}/' + artifact.name + '.zip', Buffer.from(download.data));
+            }
+      - run: |
+          set -xeo pipefail
+          pwd
+          ls
+          for archive in *.zip; do
+            name=$(basename "$archive" .zip)
```

Comment:
This basename feature is so nice, no more ${FOO%suffix} stuff. However quote the `.zip` to avoid glob expansion.

Reply:
Aren't wildcards only `*` and `?`? Can you give an example when `.zip` would be expanded?

Reply:
stupid me, ignore it. It's not a regex.